http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeoKubernetesServiceInterp.scala

```diff
@@ -0,0 +1,412 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package service
+
+import java.time.Instant
+import java.util.UUID
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.Parallel
+import cats.effect.Async
+import cats.mtl.ApplicativeAsk
+import org.broadinstitute.dsde.workbench.google2.GKEModels.{KubernetesClusterName, NodepoolName}
+import org.broadinstitute.dsde.workbench.leonardo.db.{
+  appQuery,
+  nodepoolQuery,
+  persistentDiskQuery,
+  DbReference,
+  KubernetesServiceDbQueries,
+  RuntimeServiceDbQueries,
+  SaveApp,
+  SaveKubernetesCluster
+}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.config.{
+  GalaxyAppConfig,
+  KubernetesClusterConfig,
+  NodepoolConfig,
+  PersistentDiskConfig
+}
+import org.broadinstitute.dsde.workbench.leonardo.model._
+import org.broadinstitute.dsde.workbench.leonardo.model.{LeoAuthProvider, ServiceAccountProviderConfig}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import LeonardoService.includeDeletedKey
+import io.chrisdavenport.log4cats.StructuredLogger
+import org.broadinstitute.dsde.workbench.google2.KubernetesName
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.NamespaceName
+import org.broadinstitute.dsde.workbench.leonardo.AppType.Galaxy
+import org.broadinstitute.dsde.workbench.leonardo.SamResource.PersistentDiskSamResource
+import org.broadinstitute.dsde.workbench.leonardo.http.api.{CreateDiskRequest, DiskConfigRequest}
+import org.broadinstitute.dsde.workbench.leonardo.http.service.LeoKubernetesServiceInterp.LeoKubernetesConfig
+import org.broadinstitute.dsde.workbench.leonardo.model.PersistentDiskAction.AttachPersistentDisk
+import org.broadinstitute.dsde.workbench.leonardo.model.ProjectAction.CreatePersistentDisk
+import org.broadinstitute.dsde.workbench.leonardo.service.LeoKubernetesService
+
+import scala.concurrent.ExecutionContext
+
+class LeoKubernetesServiceInterp[F[_]: Parallel](
```

Comment:
nit: could we call this just `KubernetesServiceInterp` like `DiskServiceInterp` and `RuntimeServiceInterp`?

Reply:
I avoided this initially because the the wb-libs class is also `KubernetesService`. I guess we can overload it and be explicit when the google one would conflict for consistency